File: StackKata/CSharpSolution/ZeroCapacityStack.cs
using System;

namespace StackKata {
	public class ZeroCapacityStack : IStack {

		public void push(int element) {
			throw new MyExceptions.OverflowException();
		}

		public int pop() {
			throw new MyExceptions.UnderflowException();
		}

		public int peek() {
			throw new MyExceptions.EmptyException();
		}

		public int? find(int element) {
			return null;
		}

		public Boolean isEmpty() {
			return true;
		}

		public int getSize() {
			return 0;
		}
	}
}

File: StackKata/JavaSolution/src/ZeroCapacityStack.java
public class ZeroCapacityStack implements Stack {
    @Override
    public void push(int element) {
        throw new Overflow();
    }

    @Override
    public int pop() {
        throw new Underflow();
    }

    @Override
    public int top() {
        throw new Empty();
    }

    @Override
    public int findDistanceFromTop(int element) {
        return -1;
    }

    @Override
    public int getSize() {
        return 0;
    }

    @Override
    public boolean isEmpty() {
        return true;
    }

    @Override
    public boolean isFull() {
        return true;
    }
}
